flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -317,13 +316,11 @@ public TaskManagerLocation getAssignedResourceLocation() {
                 : null;
     }
 
-    public Throwable getFailureCause() {
-        return failureCause;
-    }
-
     @Override
-    public String getFailureCauseAsString() {
-        return ExceptionUtils.stringifyException(getFailureCause());
+    public Optional<ErrorInfo> getFailureInfo() {
+        return failureCause == null
+                ? Optional.empty()
+                : Optional.of(new ErrorInfo(failureCause, getStateTimestamp(FAILED)));
```

Comment:
Does it make sense to change `failureCause` to be of type `ErrorInfo`? If yes, then we could save recreating this object multiple times.

Reply:
Yes, it does. I actually had it like that in a first version of the code but changed it for the sake of consistency after realizing that it's not Serializable (which was a problem in `ArchivedExecution`). I will revert it.